kylin/src/main/java/org/apache/zeppelin/kylin/KylinInterpreter.java

```diff
@@ -168,26 +169,31 @@ public String getSQL(String cmd) {
   private InterpreterResult executeQuery(String sql) throws IOException {
 
     HttpResponse response = prepareRequest(sql);
-
-    if (response.getStatusLine().getStatusCode() != 200) {
-      logger.error("failed to execute query: " + response.getEntity().getContent().toString());
-      return new InterpreterResult(InterpreterResult.Code.ERROR,
-          "Failed : HTTP error code " + response.getStatusLine().getStatusCode());
+    String result = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+    int code = response.getStatusLine().getStatusCode();
+    if (code != 200) {
+      StringBuilder errorMessage = new StringBuilder("Failed : HTTP error code " + code);
+      logger.error("failed to execute query: " + result);
+      try {
+        JSONObject content = new JSONObject(result);
```

Comment:
what if result is an empty string?

Reply:
If result is empty string, depending on the response code:
 if the code is 200, an empty table will be rendered;
 if the code is not 200, there will be error message, with status code, but no detailed error message, e.g. "Failed : HTTP error code 500. Error message: "